force-app/main/default/classes/SOQL.cls

```diff
@@ -271,11 +321,11 @@ public with sharing class SOQL {
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
         // Query the Account
-
+        Account account = [SELECT Id, BillingState FROM Account WHERE Id = :accountId];
         // Store the Account's BillingState
-
+        String billingState = account.BillingState;
         // Query the Contacts
-        return null; // Replace null with the variable you used to store the result
+        return [SELECT Id FROM Contact WHERE MailingState = :billingState]; // Replace null with the variable you used to store the result
```

Comment:
See if you can do this in 1 query. HINT: you can use a query within the criteria

Reply:
@cyberjus OK, so... I am probably having a "moment" here, but for the life of me I can't figure this out, how to combine into the criteria. Please share :-)